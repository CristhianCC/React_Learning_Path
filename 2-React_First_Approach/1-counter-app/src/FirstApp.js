import React from 'react';

// Functional Component
const FirstApp = () => {
    return <h1>Hello World!</h1>;
}

export default FirstApp;