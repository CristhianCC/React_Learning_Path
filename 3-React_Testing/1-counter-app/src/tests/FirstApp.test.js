import FirstApp from '../FirstApp';
import { render } from '@testing-library/react';
import { shallow } from 'enzyme';

describe('Testing FirstApp', () => {
    // test('should return Hello, Im Cristhian', () => {
    //   const greeting = 'Hello, Im Cristhian';
    //   const { getByText } = render( <FirstApp greeting={greeting} />);
    //
    //   expect(getByText(greeting)).toBeInTheDocument();
    // });

    test('should return <FirstApp /> correctly', () => {
        const greeting = 'Hello, Im Cristhian';
        const wrapper = shallow(<FirstApp greeting={ greeting } /> );

        expect(wrapper).toMatchSnapshot();
    });

    test('should show the subtitle sent by props', () => {
        const greeting = 'Hello, Im Cristhian';
        const subtitle = 'Hello, Im a subtitle';

        const wrapper = shallow(
            <FirstApp
                greeting={ greeting }
                subtitle={ subtitle }
            />
        );

        const text = wrapper.find('#text').text();

        expect(text).toBe(subtitle);
    });

});