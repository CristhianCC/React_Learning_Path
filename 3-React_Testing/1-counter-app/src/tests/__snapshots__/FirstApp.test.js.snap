// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing FirstApp should return <FirstApp /> correctly 1`] = `
<Fragment>
  <h1>
     
    Hello, Im Cristhian
     
  </h1>
  <p>
     
    I am a subtitle
     
  </p>
</Fragment>
`;
